src/backend/database_models/agent.py

```diff
@@ -0,0 +1,41 @@
+from enum import StrEnum
+
+from sqlalchemy import Enum, Float, Integer, String, Text, UniqueConstraint
+from sqlalchemy.orm import Mapped, mapped_column, relationship
+
+from backend.database_models.base import Base
+
+
+class Deployment(StrEnum):
+    COHERE_PLATFORM = "Cohere Platform"
+    SAGE_MAKER = "SageMaker"
+    AZURE = "Azure"
+    BEDROCK = "Bedrock"
+
+
+class Model(StrEnum):
+    COMMAND_R = "command-r"
+    COMMAND_R_PLUS = "command-r-plus"
+    COMMAND_LIGHT = "command-light"
+    COMMAND = "command"
+
+
+class Agent(Base):
+    __tablename__ = "agents"
+
+    version: Mapped[int] = mapped_column(Integer, default=1)
+    name: Mapped[str] = mapped_column(Text, nullable=False)
+    description: Mapped[str] = mapped_column(Text, nullable=True)
+    preamble: Mapped[str] = mapped_column(Text, nullable=True)
+    temperature: Mapped[float] = mapped_column(Float, default=0.3)
+    # tool: Mapped[List["Tool"]] = relationship()
+
+    # enum place holders
+    model: Mapped[Model] = mapped_column(Enum(Model, native_enum=False), nullable=False)
```

Comment:
OK for now: just note that we'll need to switch to  foreign keys later on, `model_deployment_id` and `model_id`